### PR TITLE
Created info commands and added user friendly command execution.

### PR DESCRIPTION
**Pull request by Duued**
Created informational commands which allows users to get information on members, users across Discord, roles, textchannels, and the guild.

Added strip_after_prefix which allows `[p] help` to work as well as `[p]help`
Added case_insensitive which allows `[p]HELP` to work as well as `[p]help`

`[p]` represents your bot prefix.

### CODE REVIEW

File: modules/info.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+from __future__ import annotations
+
+import discord
+from discord.ext import commands
+import core, typing
+#from typing import TYPE_CHECKING, Any
```
Comment (AbstractUmbra) [inline modules/info.py]: Unnecessary commented imports?

File: modules/info.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+from __future__ import annotations
+
+import discord
+from discord.ext import commands
+import core, typing
```
Comment (AbstractUmbra) [inline modules/info.py]: This isn't good practice/styling. Imports are on separated lines.

Comment (AbstractUmbra) [inline modules/info.py]: Styling decision also, we don't `import typing` but we import all necessary items `from typing`.

File: modules/info.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+class InformationEmbed(discord.Embed):
+    """A subclass of discord.Embed. 
+    
+    This class allows to automatically get information within the class instead of recreating the embed each time
+    
+    :param author: The embed author. Expects discord.Member or discord.User
+    :param entity: The Member, User, Role, TextChannel, or Guild to get information about.
+    """
+    def __init__(self, author: typing.Union[discord.Member, discord.User], entity: typing.Union[discord.Member, discord.User, discord.Role, discord.TextChannel, discord.Guild]):
```
Comment (AbstractUmbra) [inline modules/info.py]: Make these kwargs as there is a large number of parameters.

File: modules/info.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            self.set_thumbnail(url=entity.guild_avatar or entity.display_avatar or None)
+        elif isinstance(entity, discord.User):
+            description = start
+            self.set_thumbnail(url=entity.display_avatar or None)
+        elif isinstance(entity, discord.Role):
+            description = f"{start}\n\nHoisted: {entity.hoist}\n\nMentionable: {entity.mentionable}\n\n"
+        elif isinstance(entity, discord.TextChannel):
+            description = f"{start}\n\nCategory: {entity.category}\n\nNSFW: {entity.nsfw}"
+        else: # Change to elif when other types are added
+            description = f"{start}\n\nOwner: {entity.owner}"
```
Comment (AbstractUmbra) [inline modules/info.py]: Not a fan of this weird variable substitution. Can't we just:-

```suggestion
description = ""
if ...:
  description += "..."
...
```

File: modules/info.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            
+            :param entity: The user, role, or TextChannel to get information about"""
+        embed = InformationEmbed(ctx.author, entity)
+        await ctx.send(embed=embed)
+
+    @info.command(
+        name="guild",
+        brief="Get the current guild's information."
+                )
+    @commands.guild_only()
```
Comment (AbstractUmbra) [inline modules/info.py]: Definitely needs `black` ran.

**Review by AbstractUmbra — CHANGES_REQUESTED** [review]
Thanks for the PR.

Got quite a few changes here.

File: modules/info.py (hunk 1 above)
Comment (Duued) [reply modules/info.py]: Oops, forgot to remove that.  Should be fixed.

File: modules/info.py (hunk 2 above)
Comment (Duued) [reply modules/info.py]: Okay!

File: modules/info.py (hunk 3 above)
Comment (Duued) [reply modules/info.py]: If I make these kwargs how can I still ensure that the values are valid Union types?

File: modules/info.py (hunk 4 above)
Comment (Duued) [reply modules/info.py]: Working on that now.

File: launcher.py [hunk 6]
```diff
@@ -36,6 +36,8 @@ async def main() -> None:
         dsn=core.CONFIG["DATABASE"]["dsn"]
     ) as pool, LogHandler(bot=bot) as handler:
         bot.logging_queue = asyncio.Queue()
+        bot.strip_after_prefix = True
+        bot.case_insensitive = True
```
Comment (AbstractUmbra) [inline launcher.py:40]: This change is present in #52. Where does it belong?

File: modules/info.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            description += f"\n\nOwner: {entity.owner}"
+            self.set_thumbnail(url=entity.icon or None)
+
+        self.description = description
+        self.set_author(name=author.name, icon_url=author.display_avatar)
+        self.color = 0x7289DA
+
+
+class Information(core.Cog):
+    """The Mystbin file modification commands. Allows users to upload files to mystbin."""
```
[the remark this answers is not in the document]
Comment (Duued) [reply modules/info.py:72]: Lmao that was old

File: launcher.py (hunk 6 above)
Comment (Duued) [reply launcher.py:40]: Here. Deleted from #52

File: modules/info.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        aliases=["i", "info"],
+        invoke_without_command=True,
+    )
+    async def info(
+        self, ctx: core.Context, entity: discord.Member | discord.User | discord.Role | discord.TextChannel | None = None
+    ) -> None:
+        """Get information about a object
+        Args:
+            entity: The user, role, or TextChannel to get information about"""
+        embed = InformationEmbed(author=ctx.author, entity=entity or ctx.author)
```
Comment (AbstractUmbra) [inline modules/info.py:89]:
```suggestion
        self, ctx: core.Context, entity: discord.Member | discord.User | discord.Role | discord.TextChannel = commands.Author
    ) -> None:
        """Get information about a object
        Args:
            entity: The user, role, or TextChannel to get information about"""
        embed = InformationEmbed(author=ctx.author, entity=entity)
```

Comment (AbstractUmbra) [reply modules/info.py:89]: This should work but may require testing.

Comment (Duued) [reply modules/info.py:89]: I'll try it!

**Review by AbstractUmbra — APPROVED** [review]

**Pull request merged**